Gems/Terrain/Code/Source/EditorComponents/EditorTerrainPhysicsColliderComponent.cpp

```diff
@@ -14,11 +14,121 @@ namespace Terrain
+                    ->Attribute(AZ::Edit::Attributes::Category, s_categoryName)
+                    ->Attribute(AZ::Edit::Attributes::AppearsInAddComponentMenu, AZ_CRC_CE("Game"))
+                    ->Attribute(AZ::Edit::Attributes::AutoExpand, true)
+                    ->DataElement(
+                        AZ::Edit::UIHandlers::CheckBox, &EditorTerrainPhysicsColliderComponent::m_visibleInEditor, "Visible",
+                        "Always display this shape in the editor viewport.")
+                    ->DataElement(
+                        AZ::Edit::UIHandlers::Default, &EditorTerrainPhysicsColliderComponent::m_drawColor, "Color",
+                        "The color to draw the debug image.");
+            }
+        }
+    }
+
+    void EditorTerrainPhysicsColliderComponent::Activate()
+    {
+        BaseClassType::Activate();
+        AzFramework::EntityDebugDisplayEventBus::Handler::BusConnect(GetEntityId());
+        AzFramework::Terrain::TerrainDataNotificationBus::Handler::BusConnect();
+
+        GenerateDebugDrawData();
+    }
+
+    void EditorTerrainPhysicsColliderComponent::Deactivate()
+    {
+        BaseClassType::Deactivate();
+        AzFramework::Terrain::TerrainDataNotificationBus::Handler::BusDisconnect();
+        AzFramework::EntityDebugDisplayEventBus::Handler::BusDisconnect();
+    }
+
+    void EditorTerrainPhysicsColliderComponent::GenerateDebugDrawData()
+    {
+        AZStd::shared_lock<AZStd::shared_mutex> lock(m_drawMutex);
+
+        AZ::Aabb boxBounds = AZ::Aabb::CreateNull();
+        AZ::Transform transform = AZ::Transform::CreateIdentity();
+        LmbrCentral::ShapeComponentRequestsBus::Event(
+            GetEntityId(), &LmbrCentral::ShapeComponentRequestsBus::Events::GetTransformAndLocalBounds, transform, boxBounds);
+
+        int32_t gridWidth, gridHeight;
+        m_component.GetHeightfieldGridSize(gridWidth, gridHeight);
+
+        const AZ::Vector2 gridResolution = m_component.GetHeightfieldGridSpacing();
+        auto heights = m_component.GetHeights();
+
+        m_debugQuads.clear();
+        m_debugQuads.reserve((gridWidth - 1) * (gridHeight - 1));
+
+        for (int xIndex = 0; xIndex < gridWidth - 1; xIndex++)
+        {
+            for (int yIndex = 0; yIndex < gridHeight - 1; yIndex++)
+            {
+                const int index0 = yIndex * gridWidth + xIndex;
+                const int index1 = yIndex * gridWidth + xIndex + 1;
+                const int index2 = (yIndex + 1) * gridWidth + xIndex + 1;
+                const int index3 = (yIndex + 1) * gridWidth + xIndex;
+
+                const float x0 = boxBounds.GetMin().GetX() + gridResolution.GetX() * xIndex;
+                const float x1 = boxBounds.GetMin().GetX() + gridResolution.GetX() * (xIndex + 1);
+                const float y0 = boxBounds.GetMin().GetY() + gridResolution.GetY() * (yIndex + 1);
+                const float y1 = boxBounds.GetMin().GetY() + gridResolution.GetY() * yIndex;
```

Comment:
`GetMin` is called 4 times and it's not as cheap as a `GetX` or `GetY`, it'd be better to call it only once, put it into a const variable and use it.

````
const auto minBoxBounds = boxBounds.GetMin();
````